src/libraries/System.Net.Security/tests/FunctionalTests/TestConfiguration.cs

```diff
@@ -16,7 +16,7 @@ namespace System.Net.Security.Tests
     internal static class TestConfiguration
     {
         public const int PassingTestTimeoutMilliseconds = 4 * 60 * 1000;
-        public const int FailingTestTimeoutMiliseconds = 250;
+        public const int FailingTestTimeoutMiliseconds = 3 * 1000;
```

Comment:
This means that on a successful run, we'll now cause any test using this to be at least 3s instead of at least 250ms.  How commonly is this used?  Can/should we make any such tests Outerloop if they're not already?

Reply:
IMHO, any test that takes more than 1 second to run should be considered to be placed into Outerloop.

Reply:
How about 500ms to see if this is really timing problem?

To @stephentoub point, the test does not need to wait 3s
 
> System.Net.Security.Tests.ServerAsyncAuthenticateTest.ServerAsyncAuthenticate_MismatchProtocols_Fails(serverProtocol: Tls, clientProtocol: Tls11, expectedException: typeof(System.Security.Authentication.AuthenticationException)) [FINISHED] Time: 0.0025739s

However there are some which do. If we still see failures I think I can rework the helper structure little bit more to fail quickly.

Reply:
> To @stephentoub point, the test does not need to wait 3s

Then there's a naming issue here.  There are two kinds of timeouts we see in our tests:
- a timeout when we expect something to succeed and we want to kill the test if it takes too long.  We've named this "passing".
- a timeout when we expect something to fail or hang indefinitely, but since we can't wait forever, we substitute some reasonably long timeout that we treat as "if it waited this long, we'll consider that infinite and thus the test passes".  We've naming this "failing".

So there's no way we should be using a "failing" test timeout of 3s and have the test successfully complete in .002 seconds.

Reply:
In this case the pass/fail expectation is passed in to the helper as bool and is related to the ssl handshake it self. When handshake is expected to fail, it can fail in (at least) three different ways: AuthenticationException e.g. peers agree that there is something wrong (fast), IOException when one side closes connection on error (fast) or nothing/timeout (slow)

I'm open to suggestions about names. Part of the problem may be that the `ServerAsyncSslHelper()` blends all three failures together has single timeout for all of them

Note that the time reported by XUnit is whole test execution, while the timeout is just for the handshake it self.

Reply:
I look at the ServerAsyncAuthenticate_MismatchProtocols_Fails once again. It turns out that in case of Ssl2/3 the client throws and fails to send anything. From ServerAsyncAuthenticateTest prospective it is completely useless IMHO as no sever aspect of SslStream is never touched. Effectively this is just Sleep() in test execution.

Reply:
I rework this more and and I don't see any slow tests any more. I set locally FailingTestTimeoutMiliseconds to 20s and run whole suite. 
This is done by excluding protocols not supported by given platform and by adding Ssl3 alert. 
That leaves Windows 7/8 with .5s timeout for Sslv2.